### PR TITLE
Fixes #14399 - fix error on inc update with composites

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
Also adds support for doing an incremental update without specifying an environment

**Comment by beav** [comment]
@jlsherrill a newbie question, but I am having some trouble getting the incremental update button to appear in the web ui to test this PR. How do you usually kick off the incremental update?

**Comment by bbuckingham** [comment]
@beav, The incremental update is initiated through the Content -> Errata page.

The following is an example scenario:
- go to Content -> Errata
- select an applicable errata
- select Content Hosts tab
- select a host
- select Apply to Hosts
- select 'Apply to Content Host Immediately after publishing'
- click Confirm

I had tested out a couple of the other bug fixes for incremental updates, so I can try this one out as well.

**Comment by beav** [comment]
@bbuckingham thanks for the info, and for covering this PR :koala:

**Comment by bbuckingham** [comment]
I have tested this PR using the scenario described above; however, I am not seeing the expected behavior based on interpretation of original bug description.  

One note regarding the scenario tested, the content host was registered to a component view (e.g. zoo 1.0) that needed an incremental update.  That component view (zoo 1.0) was also part of a published composite view (e.g. composite 1.0).  

In discussing the original bug with QE, the assumption was that performing an incremental update on the component (zoo 1.0), would also propagate the incremental update to the composite; however, I do not see this occuring.  The following is an example of how this could be done with the cli:
```
hammer> content-view version incremental-update --lifecycle-environment-ids 2 --content-view-version-id 12 --propagate-all-composites true --puppet-module-ids 1

```

**Comment by jlsherrill** [comment]
@bbuckingham i'm seeing it work for me, did you use this PR:  https://github.com/Katello/hammer-cli-katello/pull/376 as well?  Apologies for not including it in my original description.  

With those two prs i'm seeing:

```
[vagrant@robot hammer-cli-katello]$ hammer -u admin -p changeme  content-view version incremental-update --lifecycle-environment-ids 1 --content-view-version-id 57  --propagate-all-composites true  --puppet-module-ids=1
[.................................................................................................................................] [100%]
Content View: puppet-component version 1.1
Added Content:
  Puppet Modules:
        puppetlabs-apache-1.3.0

Content View: puppet-composite version 1.1
Added Content:
  Puppet Modules:
        puppetlabs-apache-1.3.0
```

**Comment by jlsherrill** [comment]
@bbuckingham actually that pr shouldn't be required (its only required if you don't want to specify an environment id which has nothing to do with composites).  I can't seem to reproduce the issue, can you?

File: app/lib/actions/katello/content_view_version/incremental_update.rb [hunk 1]
```diff
@@ -108,7 +108,7 @@ def finalize
         # given a composite version, and a list of new components, calculate the list of all components for the new version
         def calculate_components(old_version, new_components)
           old_components = old_version.components.select do |component|
-            !old_version.components.map(&:content_view_id).include?(component.content_view_id)
+            !new_components.map(&:content_view_id).include?(component.content_view_id)
           end
```
Comment (jlsherrill) [inline app/lib/actions/katello/content_view_version/incremental_update.rb]: @bbuckingham this is the change from what we just discovered.  If you want me to walk you through what this is doing and explain why it was wrong, i'd be happy to!

Comment (bbuckingham) [inline app/lib/actions/katello/content_view_version/incremental_update.rb]: @jlsherrill, With this change, I am now able to perform incremental update on a content host registered to a composite view (with multiple components).

**Comment by bbuckingham** [comment]
ACK

**Pull request merged**